### PR TITLE
Add SiteList actions to active statuses in ReqMgr2

### LINKED ISSUE #12037: Support change to SiteWhitelist/SiteBlacklist in ReqMgr2 for active workflows
**Impact of the new feature**
ReqMgr2

**Is your feature request related to a problem? Please describe.**
This is a sub-task of this meta-issue: https://github.com/dmwm/WMCore/issues/8323
Towards providing a feature that allows workflow site lists to be changed while workflows are active in the system.

**Describe the solution you'd like**
With this ticket, we are supposed to implement changes to ReqMgr2 such that `SiteWhitelist` and `SiteBlacklist` are allowed to be updated in workflows that are in one of the given statuses: 
```
staging, acquired, running-open
```

Note that this change needs to be persisted in two places:
* workflow high level description (json)
* workflow spec file (the WMWorkload object file)

If possible, we should also validate whether the sitelists is actually changing or not. If none of them change, then nothing should be persisted in the database and no extra actions need to be triggered. 

The reason we are not going to accept these changes for workflows in `assigned` and `staged` is such that we can guarantee no data race between services dealing with the same workflow.

An initial point for this development will have to be made here:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/DataStructs/RequestStatus.py#L118

**Describe alternatives you've considered**
Unless explicitly requested, I don't think we need to add these capabilities to the ReqMgr2 UI.

**Additional context**
None

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #12037 

#### Status

**SUPERSEEDED BY: https://github.com/dmwm/WMCore/pull/12148**

#### Description
With the current change we allow SiteLists related actions for statuses: staging, acquired, running-open in ReqMgr2

Before making a call to central services for changing any of the request parameters, an additional step is executed to
to check which are the allowed parameter modifications for the given status and if the so provided new values fromthe rest call actually differ from the workflow parameters already defined in central couchdb.

With the current change we no longer ignore all the rest of the request arguments provided with the REST call in the cases of a change of the Request priority. See: https://github.com/dmwm/WMCore/issues/8457#issuecomment-366195086

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
This PR provides a cherry-pick of 3 pull requests that have been recently reverted. Changes have been originally provided by:
https://github.com/dmwm/WMCore/pull/12077
https://github.com/dmwm/WMCore/pull/12108
https://github.com/dmwm/WMCore/pull/12111

#### External dependencies / deployment changes
None

### CODE REVIEW

**Comment by amaltaro** [comment]
@todor-ivanov from our Monday meeting, I had understood this had already been taken care of, but I failed to find any relevant PR and/or the current ones you have opened don't include these changes, so here we go.

Can you please double-check it to make sure I am not missing anything? In addition, feel free to change anything in this PR and/or push more changes into it.

Ideally we could add the developments you made in https://github.com/dmwm/WMCore/pull/12120 into this one (or rebase 12120 on top of these changes). Validating and letting all these changes to go together.

**Comment by todor-ivanov** [comment]
hi @amaltaro  
I can see the benefit of accumulating the full set of code changes related to the original issue in a single PR. But I am afraid checrry-picking   alredy merged PRs/commits would mess history and may crate code conflicts. 

Actually the PR which is fixing your latest reported bug is https://github.com/dmwm/WMCore/pull/12120 , which is indeed missing from this set here. So What I'd do is to complete the tests requested by you from here: https://github.com/dmwm/WMCore/pull/12120#pullrequestreview-2334263034  (actually repeat them again, because last time I did it through an interactive on a previously stuck workflows already present in the system, rather than with dedicated  workflows but anyway ...)  And upon that just merge the PR #12120 which should properly resolve the original issue.

**Comment by amaltaro** [comment]
@todor-ivanov your PR #12120 is inconsistent and it needs to be rebased, reason is because you created it when we had those 3 PRs already merged in (which are provided again in this PR). After reverting those 3 PRs last week, your 2 PRs became inconsistent, as some codebase that it was relying on no longer exist in master/HEAD.

I guess the tests you are running at the moment are based on 2.3.6rc7 + your #12120 changes. If you were to use master, you would not even be able to patch things out. Have a look at this file to better understand what went in and in which release:
https://github.com/dmwm/WMCore/blob/master/CHANGES

Bottom line, these changes are needed in the repository and for your PRs. There are multiple ways to move forward on this, as said in my previous message, so I let you decide.

By the way, if you feel like squashing those commits properly, I think it would be beneficial as well - not mandatory though.

**Comment by todor-ivanov** [comment]
hi @amaltaro  I realized  all that after writing my comment above... And I am fighting with the already messed up repo history for quite some time already today .... And the reason being that those PRs of mine have been reverted with separate commits ... instead of rebasing the master branch etc. etc. which would have given the chance for this whole development to be in a separate branch ... , while  the tag 2.3.6 to be still cut on top of  master..... anyway ... we have what we have. I am working on ramping it ll up to the relevant version. I can either rebase the latest PR #12120  on top of your branch ... but then it will mix mine and your repo, which I am not a big fan of. Or I can repeat all your actions with cherripicking  of the previous commits and rebase only inside my local tree and then create  yet another PR (Which I prefer btw), this will help me isolate the whole work only in my repository, so I'll have the full freedom to work instead of waiting for you.

**Comment by amaltaro** [comment]
Rebasing the master branch would have caused change of history, so definitely a not-recommended action.
Yes, maybe cherry-picking these commits in a new PR and applying your latest changes on top of it are the way to go.
If you do that, feel free to close this one out. Thanks

**Comment by todor-ivanov** [comment]
**NEVER MERGE**

**SUPERSEDED by:  https://github.com/dmwm/WMCore/pull/12148**

**Comment by todor-ivanov** [comment]
Closing it without merge
